src/hooks/sounds.tsx

```diff
@@ -35,35 +37,56 @@ export interface ExposedDataWithPlay extends ExposedData {
 
 export interface CustomSoundsState {
   soundConfigs: SoundConfigMap;
+  isLoading: boolean;
 }
 
 export const initialValue: CustomSoundsState = {
   soundConfigs: {},
+  isLoading: false,
 };
 
 export const CustomSoundsContext = createContext<CustomSoundsState>(
   initialValue
 );
 
-export const CustomSoundsProvider: React.FC = ({ children }) => {
+export type CustomSoundsType = {
+  loadingComponent: React.ReactNode;
+  waitTillConfigLoaded: boolean;
+};
+
+export const CustomSoundsProvider: React.FC<CustomSoundsType> = ({
+  loadingComponent,
+  waitTillConfigLoaded,
+  children,
+}) => {
   const [soundConfigs, setSoundConfigs] = useState<SoundConfigMap>(
     initialValue.soundConfigs
   );
+  const [isLoading, setIsLoading] = useState<boolean>(initialValue.isLoading);
+  const { user } = useUser();
+  const userId = user?.uid;
 
   // Fetch the sound configs data on first load
   useEffect(() => {
-    fetchSoundConfigs().then((soundConfigs) => {
-      setSoundConfigs(soundConfigs);
-    });
-  }, []);
+    if (!userId) return;
+    setIsLoading(true);
+    fetchSoundConfigs()
+      .then((soundConfigs) => setSoundConfigs(soundConfigs))
+      .finally(() => setIsLoading(false));
```

Comment:
[NB] I think you could just use `setSoundConfigs` as the 'handler' in the `.then()` directly:

```suggestion
    fetchSoundConfigs()
      .then(setSoundConfigs)
      .finally(() => setIsLoading(false));
```